BattleTank/Input/GenericGamepadTankActionProvider.cs

```diff
@@ -5,38 +5,54 @@
 using System.Text;
 using System.Threading.Tasks;
 using SlimDX.DirectInput;
+using Microsoft.Xna.Framework.Input;
+using Microsoft.Xna.Framework;
 
 namespace BattleTank.Input
 {
     class GenericGamepadTankActionProvider : ITankActionProvider
     {
         private const int MAX_AXIS_VALUE = 5000;
 
-        public static GenericGamepadTankActionProvider DefaultGamepadProvider { get; } = new GenericGamepadTankActionProvider(2, 0, 7);
-
+        public static GenericGamepadTankActionProvider DefaultPlayerOneGamepadProvider { get; } = new GenericGamepadTankActionProvider(2, 0, 7);
+        public static GenericGamepadTankActionProvider DefaultPlayerTwoGamepadProvider { get; } = new GenericGamepadTankActionProvider(2, 0, 7);
         public int SpeedBoostButtonNumber { get; set; }
         public int PlantMineButtonNumber { get; set; }
         public int FireButtonNumber { get; set; }
 
-        private Joystick _joystick;
+        private SlimDX.DirectInput.Joystick _joystick;
 
         public GenericGamepadTankActionProvider(int speedBoostButtonNumber, int plantMineButtonNumber, int fireButtonNumber)
         {
             SpeedBoostButtonNumber = speedBoostButtonNumber;
             PlantMineButtonNumber = plantMineButtonNumber;
             FireButtonNumber = fireButtonNumber;
 
-            Initialize();
+            Initialize(1);
+            Initialize(2);
         }
 
-        private bool Initialize()
+        private bool Initialize(int pad)
         {
             DirectInput dinput = new DirectInput();
-            DeviceInstance devices = dinput.GetDevices(DeviceClass.GameController, DeviceEnumerationFlags.AttachedOnly).FirstOrDefault();
 
-            if (devices is null) return false;
+        
+            if (pad == 1)
+            {
+                DeviceInstance device1 = dinput.GetDevices(DeviceClass.GameController, DeviceEnumerationFlags.AttachedOnly).FirstOrDefault();
+                if (device1 is null) return false;
 
-            _joystick = new Joystick(dinput, devices.InstanceGuid);
+                _joystick = new SlimDX.DirectInput.Joystick(dinput, device1.InstanceGuid);
+            
+            }
+            if (pad == 2)
+            {
+                DeviceInstance device2 = dinput.GetDevices(DeviceClass.GameController, DeviceEnumerationFlags.AttachedOnly).LastOrDefault();
```

Comment:
Zobacz @filerd002 ta linijka i ta wyżej są prawie dokładnie takie same, różnią się tylko końcówą FirstOrDefault(), LastOrDefault(). Nie byłoby czytelniej gdyby zapisać list z GetDevices do zmiennej i potem ją współdzielić. Przy okazji ograniczasz ilość zapytań do urządzenia, a nuż może się zdażyć że na jakimś kompie będzie to bardzo wolne :).

Reply:
Dasz rade to zmodyfikować na szybko? :D

Reply:
Bo jak nie to jutro się tym zajmę

Reply:
Dobra, jutro rano w pracy to ogarnę bo dzis i jutro mam luźniej :).

Reply:
:+1: